src/main/java/org/elasticsearch/index/query/RangeQueryBuilder.java

```diff
@@ -41,385 +53,215 @@
 
     private boolean includeUpper = true;
 
-    private float boost = -1;
+    private float boost = 1.0f;
 
     private String queryName;
 
+    private String format;
+
     /**
      * A Query that matches documents within an range of terms.
      *
-     * @param name The field name
+     * @param fieldname The field name
      */
-    public RangeQueryBuilder(String name) {
-        this.name = name;
+    public RangeQueryBuilder(String fieldname) {
+        this.fieldname = fieldname;
     }
 
-    /**
-     * The from part of the range query. Null indicates unbounded.
-     */
-    public RangeQueryBuilder from(Object from) {
-        this.from = from;
-        return this;
+    public RangeQueryBuilder() {
+        // for serialization
     }
 
     /**
-     * The from part of the range query. Null indicates unbounded.
+     * Get the field name for this query.
      */
-    public RangeQueryBuilder from(String from) {
-        this.from = from;
-        return this;
+    public String fieldname() {
+        return this.fieldname;
     }
 
     /**
      * The from part of the range query. Null indicates unbounded.
      */
-    public RangeQueryBuilder from(int from) {
-        this.from = from;
-        return this;
-    }
-
-    /**
-     * The from part of the range query. Null indicates unbounded.
-     */
-    public RangeQueryBuilder from(long from) {
-        this.from = from;
-        return this;
-    }
-
-    /**
-     * The from part of the range query. Null indicates unbounded.
-     */
-    public RangeQueryBuilder from(float from) {
-        this.from = from;
+    public RangeQueryBuilder from(Object from, boolean includeLower) {
```

Comment:
I like this simplication, this is a breaking change though, we need to note this down somewhere, for instance in the migrate_2.0 guide. Actually we could have our own specific file since we don't know yet if this will make it for 2.0.